### PR TITLE
Enhancement: Metrics for Search Took Times using Action Listeners

### PR DESCRIPTION
**Pull request by JVerwolf**
This PR instruments the took times for search responses.  It does this by adding instrumentation to the `TransportSearchAction` and the `TransportSearchScrollAction`'s doExecute method.

This PR supersedes the prior PR which attempted do add instrumentation for the took times in the construction of the Search Response class: https://github.com/elastic/elasticsearch/pull/104292

The changes required are in:
- server/src/main/java/org/elasticsearch/action/search/`TransportSearchAction`.java
- server/src/main/java/org/elasticsearch/action/search/`TransportSearchScrollAction`.java

The following Actions do not require modification, as they simply translate to calls to the above actions. We don't want to record the same metric twice, as this would skew the times reported.
- server/src/main/java/org/elasticsearch/action/search/`TransportMultiSearchAction`.java
- x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/`EnrichCoordinatorProxyAction`.java
- x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/action/`TransportQuerySearchApplicationAction`.java

Thanks to @quux00 for suggesting this much simpler approach!

### CODE REVIEW

File: modules/parent-join/src/internalClusterTest/java/org/elasticsearch/join/query/ChildQuerySearchIT.java [hunk 1]
```diff
@@ -1404,6 +1404,7 @@ public void testParentChildQueriesViaScrollApi() throws Exception {
 
         for (QueryBuilder query : queries) {
             assertScrollResponsesAndHitCount(
+                client(),
```
Comment (JVerwolf) [inline modules/parent-join/src/internalClusterTest/java/org/elasticsearch/join/query/ChildQuerySearchIT.java:1407]: I updated the `assertScrollResponsesAndHitCount`  test helper method so that it will work with the ESSingleNodeTestCase, and thus make it available for testing the metric changes.  Hence this seemingly unrelated change.

**Review by piergm — APPROVED** [review]
Looks really good to me @JVerwolf!

**Review by quux00 — APPROVED** [review]
LGTM!
